fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java

```diff
@@ -381,4 +390,303 @@ void setProjectVariables(List<OToolVariable> variables) {
         }
     }
 
+    private static <T> Optional<T> findConfig(Set<T> configs, Predicate<T> predicate) {
+        return configs.stream()
```

Comment:
this is huge. This really s hold go to its own class with all the logic.
Some ReverseProjectSearch. even if it wouldbe jsut utility class. And of course  such class will deserver some tests

Reply:
it's not that huge? do you insist on split?

Reply:
it add 320lines to class of 150lines. judge on your own